FutureNHS/futurenhs-platform
# Create shared dev Key Vault, add 'Sealed Secrets' Certificate

![jason-pofahl-6AQY7pO1lS0-unsplash](https://user-images.githubusercontent.com/13286587/87547827-929db780-c6a3-11ea-9276-8eaf30325894.jpg)

[Kanban board ticket](https://airtable.com/tblgtzV3sSwquTkIa/viw8ZHoNqvSvl62lt/rec9fJAh3AkmbD7KW?blocks=hide)

- This PR creates the long term infrastructure and certificate neccessary for the [Sealed Secrets controller](https://engineering.bitnami.com/articles/sealed-secrets.html) to decrypt `Sealed Secrets`in this repository, into standard Kubernetes `Secrets`.
- For developer convenience, we store this shared certificate in a long living[Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/), named `fnhs-shared-dev`, which all will use whilst developing.
- By retrieving this secret and adding to their development cluster, the`sealed secret controller` running within the developer cluster is able to unseal any `sealed secrets` in this repository.


### How do I make use of this?
- Create your development cluster as normal.
- Fetch the Sealed Secret secret from the shared dev key vault with the following command:
`az keyvault secret show --vault-name "fnhs-shared-dev" --name "sealed-secret-yaml"`.
- Copy the 'value' that is printed from the above command into a yaml file, and apply to your cluster with the following command:
`(cd infrastructure/environments/shared-dev && kubectl apply -f sealed-secret-cert.yaml)`

## Acceptance Criteria
- [x] A developer is able to access the shared certificate from the shared Key Vault, and add it to their cluster.

NOTE: [A follow up PR](https://github.com/FutureNHS/futurenhs-platform/pull/59) adds the sealed secret controller to the dev cluster and will allow sealing and unsealing of Sealed Secrets.

## Pre-review checklist:



- [ ] Analytics (user analytics, e.g. Google Analytics)

- [ ] Observability (metrics/tracing)


## Testing information

To test this PR, do the following:

- Retrieve the secret from the Azure Key Vault.
- Apply the secret, and verify it has successfully been added to your cluster.


## Test:


- Platforms/Browsers: 

  - [ ] Google Chrome (latest version) 
  
  
  
- Accessibility:

  
  
  - [ ] Magnification
  
  
  - [ ] Text size 200%
  
  - [ ] Touch target areas (Mobile)
  
  - [ ] Landscape (Mobile)
